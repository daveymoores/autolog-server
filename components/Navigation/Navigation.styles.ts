import css from "styled-jsx/css";

export default css`
  nav {
    display: flex;
    justify-content: space-between;
    align-items: stretch;

    max-width: var(--max-content-width);
    width: 100%;
    height: var(--cellHeight);
    border: var(--lineWidth) solid var(--lightGreen);
    margin: 0 auto;
  }

  @media (prefers-color-scheme: dark) {
    nav {
      border: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    nav {
      border: var(--lineWidth) solid var(--darkGrey);
    }
  }

  nav > span {
    border-top: none;
    border-bottom: none;
    border-left: none;
    font-size: 18px;
    font-weight: 700;
    letter-spacing: 0.05em;
  }

  @media screen and (max-width: 501px) {
    nav > span {
      border-right: none;
    }
  }

  ul {
    display: flex;
    justify-content: center;
    align-items: stretch;

    list-style: none;

    padding: 0;
    margin: 0;
  }

  li {
    border-top: none;
    border-bottom: none;
    border-right: none;
  }
`;
