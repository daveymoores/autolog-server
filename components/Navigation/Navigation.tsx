import Image from "next/image";
import Link from "next/link";
import React from "react";

const Navigation: React.FC = () => {
  return (
    <div className="container grid grid-cols-12 mt-8 lg:mt-16 mx-auto lg:max-w-screen-xl font-semibold">
      <div className="col-span-12 md:col-span-10 md:col-start-2 lg:col-span-8 lg:col-start-3">
        <nav className="flex justify-between align-middle">
          <Link href="/" passHref>
            <a>
              <Image
                src="/logo.svg"
                width={142}
                height={30}
                priority={true}
                layout="fixed"
              />
            </a>
          </Link>
          <Link href="/documentation">Documentation</Link>
        </nav>
      </div>
    </div>
  );
};

export default Navigation;
