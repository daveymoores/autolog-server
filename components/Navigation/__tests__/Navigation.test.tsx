import { render, screen } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";

import Navigation from "../Navigation";

describe("Navigation", () => {
  it("renders a title", () => {
    render(<Navigation />);

    const nav = screen.getByRole("navigation");

    expect(nav).toBeInTheDocument();
  });

  it("renders navigation unchanged", () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
