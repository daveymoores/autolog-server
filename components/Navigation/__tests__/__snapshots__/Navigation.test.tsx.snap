// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders navigation unchanged 1`] = `
<div
  className="container grid grid-cols-12 mt-8 lg:mt-16 mx-auto lg:max-w-screen-xl font-semibold"
>
  <div
    className="col-span-12 md:col-span-10 md:col-start-2 lg:col-span-8 lg:col-start-3"
  >
    <nav
      className="flex justify-between align-middle"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 15,
              "margin": 0,
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": 114,
            }
          }
        >
          <img
            data-nimg="fixed"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="/logo.svg"
            srcSet="/logo.svg 1x, /logo.svg 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </span>
      </a>
      <a
        href="/documentation"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Documentation
      </a>
    </nav>
  </div>
</div>
`;
